bowtie/_cli.py

```diff
@@ -1684,6 +1708,623 @@ async def info(
+
+        with progress:
+            responses = await asyncio.gather(
+                *[
+                    download_and_parse_latest_report_for(dialect)
+                    for dialect in dialects
+                ],
+            )
+
+            progress.update(
+                task,
+                description=(
+                    f"Successfully downloaded and parsed all latest "
+                    f"reports of {id} for {pretty_names_str}!"
+                ),
+                completed=total_files,
+                total=total_files,
+            )
+            return responses
+
+
+def _trend_table_for(
+    id: ConnectableId,
+    versions: Set[str],
+    dialects_trend: dict[Dialect, _report.Report],
+) -> Table:
+    main_table = Table(show_lines=True)
+    main_table.add_column(
+        "Dialect",
+        justify="center",
+        vertical="middle",
+    )
+    main_table.add_column(
+        f"Trend Data of {id} versions",
+        justify="center",
+    )
+
+    def add_sub_table_row(
+        sub_table: Table,
+        version: str,
+        unsuccessful: Unsuccessful,
+    ):
+        sub_table.add_row(
+            version,
+            str(len(unsuccessful.skipped)),
+            str(len(unsuccessful.errored)),
+            str(len(unsuccessful.failed)),
+        )
+
+    for dialect, report in dialects_trend.items():
+        test = "tests" if report.total_tests != 1 else "test"
+        sub_table = Table(
+            "Version",
+            "Skips",
+            "Errors",
+            "Failures",
+            caption=f"{report.total_tests} {test} ran",
+            show_edge=False,
+        )
+
+        if versions:
+            for version, unsuccessful in report.latest_to_oldest():
+                add_sub_table_row(sub_table, version, unsuccessful)
+        else:
+            implementation = report.implementations[id]
+            version = implementation.version or "latest"
+            unsuccessful = report.unsuccessful(id)
+            add_sub_table_row(sub_table, version, unsuccessful)
+
+        main_table.add_row(dialect.pretty_name, sub_table)
+
+    return main_table
+
+
+def _trend_table_in_markdown_for(
+    id: ConnectableId,
+    versions: Set[str],
+    dialects_trend: dict[Dialect, _report.Report],
+) -> str:
+    rows_data: list[list[str]] = []
+
+    inner_table_columns = [
+        "Version",
+        "Skips",
+        "Errors",
+        "Failures",
+    ]
+
+    def create_inner_table_row(
+        version: str,
+        unsuccessful: Unsuccessful,
+    ):
+        return [
+            version,
+            str(len(unsuccessful.skipped)),
+            str(len(unsuccessful.errored)),
+            str(len(unsuccessful.failed)),
+        ]
+
+    for dialect, report in dialects_trend.items():
+        test = "tests" if report.total_tests != 1 else "test"
+        inner_table_rows: list[list[str]] = []
+        if versions:
+            for version, unsuccessful in report.latest_to_oldest():
```

Comment:
Make this a list comprehension, especially once you remove the other branch, which then will let you delete the `create_inner_table_row` function and just do this all inline.

Reply:
I think the fix for this would also get easier and the reasoning for that is the same as the above comment.